client/beefy/src/worker.rs

```diff
@@ -207,16 +228,49 @@ pub(crate) struct WorkerParams<B: Block, BE, C, P, R, N> {
 	pub on_demand_justifications: OnDemandJustificationsEngine<B, R>,
 	pub links: BeefyVoterLinks<B>,
 	pub metrics: Option<Metrics>,
-	pub min_block_delta: u32,
+	pub persisted_state: PersistedState<B>,
+}
+
+#[derive(Debug, Decode, Encode, PartialEq)]
+pub(crate) struct PersistedState<B: Block> {
+	/// Best block we received a GRANDPA finality for.
+	best_grandpa_block_header: <B as Block>::Header,
+	/// Best block a BEEFY voting round has been concluded for.
+	best_beefy_block: NumberFor<B>,
+	/// Chooses which incoming votes to accept and which votes to generate.
+	/// Keeps track of voting seen for current and future rounds.
+	voting_oracle: VoterOracle<B>,
+}
+
+impl<B: Block> PersistedState<B> {
+	pub fn initialize(
+		grandpa_header: <B as Block>::Header,
+		best_beefy: NumberFor<B>,
+		session_start: NumberFor<B>,
+		validator_set: ValidatorSet<AuthorityId>,
+		min_block_delta: u32,
+	) -> Self {
+		PersistedState {
+			best_grandpa_block_header: grandpa_header,
+			best_beefy_block: best_beefy,
+			voting_oracle: VoterOracle::genesis(session_start, validator_set, min_block_delta),
+		}
+	}
+
+	pub(crate) fn set_min_block_delta(&mut self, min_block_delta: u32) {
+		self.voting_oracle.min_block_delta = min_block_delta;
```

Comment:
Just safety question - `VoterOracle::genesis()` is using `1` if `min_block_delta` is `0` (`min_block_delta: min_block_delta.max(1)`) and here we don't check this (and it is called from the `load_persistent` where `BeefyParams::min_block_delta` is directly passed as an argument). Can it break something?

Also it seems that the `min_block_delta` is not actually persistent, since it is changed when we load state from the db. But it is fine to leave as is - just mumbling :)